src/Services/CodeParser/CodeParser.php

```diff
@@ -77,16 +84,16 @@ public function getMethodCalls(
                 return false;
             }
 
-            // Check if call matches what we're looking for ('like 'dispatch')
+            // Check if call matches what we're looking for (like 'dispatch')
             if ($node->name->toString() !== $methodName) {
                 return false;
             }
 
             // This happens when you inject the dispatcher and do '$dispatcher->dispatch()'
             if ($node->var instanceof Variable) {
-                $variableClass = $this->resolveClassFromVariable($node->var);
+                $variableClasses = $this->resolveClassesFromVariable($node->var);
 
-                return $this->areClassesSame($variableClass, $subjectClass);
+                return count($variableClasses) === 1 && $this->areClassesSame($variableClasses[0], $subjectClass);
```

Comment:
@JonasPardon this is for backwards-compatibility, but you may have more context around the use cases here.